scripts/comfyui.py

```diff
@@ -49,45 +48,61 @@ def process(self, p, *args):
 
         self.outpath_samples = p.outpath_samples
 
-    def postprocess(self, p, res, queue_front, output_node_label, **kwargs):
+    def postprocess_batch(self, p, queue_front, output_node_label, images: list, batch_number, **kwargs):
         if not getattr(shared.opts, 'comfyui_enabled', True):
             return
+        if getattr(shared.state, 'interrupted', False):
+            return
+        if len(images) == 0:
+            return
 
-        images = res.images[res.index_of_first_image:]
-        results = res.images[:res.index_of_first_image]
-        initial_amount_of_images = len(images)
-        for i in range(p.n_iter):
-            if getattr(shared.state, 'interrupted', False):
-                return
-            range_start = i*p.batch_size
-            range_end = (i+1)*p.batch_size
-            images_batch = images[range_start:range_end]
-            batch_results = ComfyuiNodeWidgetRequests.start_workflow_sync(
-                batch=images_batch,
-                workflow_type='postprocess',
-                is_img2img=self.is_img2img,
-                required_node_types=webui_postprocess_output.expected_node_types,
-                queue_front=queue_front,
-            )
-
-            if batch_results is None or 'error' in batch_results:
-                continue
-
-            amount_of_images_per_image = len(batch_results)
-            for j, image in enumerate(images_batch):
-                for k in range(amount_of_images_per_image):
-                    batch_results[j*amount_of_images_per_image + k].info = images_batch[j].info
-
-            results.extend(batch_results)
+        batch_results = ComfyuiNodeWidgetRequests.start_workflow_sync(
+            batch=images,
+            workflow_type='postprocess',
+            is_img2img=self.is_img2img,
+            required_node_types=webui_postprocess_output.expected_node_types,
+            queue_front=queue_front,
+        )
 
-        batch_count_multiplier = (len(results) - res.index_of_first_image) // initial_amount_of_images
-        p.n_iter = p.n_iter * batch_count_multiplier
-        res.images = results
+        if batch_results is None or 'error' in batch_results:
+            return
 
-        self.save_image(results)
+        batch_size_factor = len(batch_results) // len(images)
+
+        if batch_number == 0:
+            lists_to_scale = [p.prompts, p.negative_prompts, p.seeds, p.subseeds,
+                              p.all_prompts, p.all_negative_prompts, p.all_seeds, p.all_subseeds]

```

Comment:
This does not work as far as my testing goes. Rescaling `p.all_*` breaks generation and un-syncs the seeds.

To reproduce the issue, run a generation with batch size = 2, output nodes count = 3 and batch count = 5 (or any other combination of prime numbers)

Reply:
how do we fix this then?

Reply:
gotta patch create_infotext. See my PR for an implementation:
https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/ca45ff1ae6fdd5c2dcd754fde95dd29f49bd414b/modules/processing.py#L720-L738